File: src/main/java/io/endertech/util/helper/LogHelper.java
package io.endertech.util.helper;

import io.endertech.reference.Reference;
import org.apache.logging.log4j.Level;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class LogHelper
{
    private static final Logger logger = LogManager.getLogger(Reference.MOD_ID);

    public static void log(Level logLevel, String format, Object... data) { logger.log(logLevel, format, data); }

    public static void debug(String format, Object... data)
    {
        log(Level.DEBUG, format, data);
    }

    public static void error(String format, Object... data)
    {
        log(Level.ERROR, format, data);
    }

    public static void fatal(String format, Object... data)
    {
        log(Level.FATAL, format, data);
    }

    public static void info(String format, Object... data)
    {
        log(Level.INFO, format, data);
    }

    public static void warn(String format, Object... data)
    {
        log(Level.WARN, format, data);
    }
}

File: src/main/java/io/endertech/util/helper/LogHelper.kt
package io.endertech.util.helper

import io.endertech.reference.Reference
import org.apache.logging.log4j.Level
import org.apache.logging.log4j.LogManager

object LogHelper {
    private val logger = LogManager.getLogger(Reference.MOD_ID)

    fun log(logLevel: Level, format: String, vararg data: Any) {
        logger.log(logLevel, format, *data)
    }

    fun debug(format: String, vararg data: Any) {
        log(Level.DEBUG, format, *data)
    }

    fun error(format: String, vararg data: Any) {
        log(Level.ERROR, format, *data)
    }

    fun fatal(format: String, vararg data: Any) {
        log(Level.FATAL, format, *data)
    }

    fun info(format: String, vararg data: Any) {
        log(Level.INFO, format, *data)
    }

    fun warn(format: String, vararg data: Any) {
        log(Level.WARN, format, *data)
    }
}
